src/benchmark/file_io_read_micro_benchmark.cpp

```diff
@@ -16,16 +16,40 @@
 
 namespace hyrise {
 
-void read_data_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start) {
+
+void read_data_using_read(const size_t from, const size_t to, int32_t fd, uint32_t* read_data_start,
+                          std::atomic<bool>& threads_ready_to_be_executed) {
+  while (!threads_ready_to_be_executed) {}
   const auto uint32_t_size = ssize_t{sizeof(uint32_t)};
-  const auto bytes_to_read = static_cast<ssize_t>(uint32_t_size * (to - from));
-  lseek(fd, from * uint32_t_size, SEEK_SET);
-  Assert((read(fd, read_data_start + from, bytes_to_read) == bytes_to_read),
-         close_file_and_return_error_message(fd, "Read error: ", errno));
+  const auto total_bytes_to_read = static_cast<ssize_t>(uint32_t_size * (to - from));
+  const auto elements_to_read = static_cast<uint64_t>(total_bytes_to_read / uint32_t_size);
+  const auto MAX_NUMBER_OF_ELEMENTS = uint64_t{536'869'888};
+
+  if (elements_to_read > MAX_NUMBER_OF_ELEMENTS) {
+    auto elements_read = uint64_t{0};
+    auto elements_remaining = elements_to_read;
+    while (elements_remaining > 0) {
+      lseek(fd, (from + elements_read) * uint32_t_size, SEEK_SET);
+      auto elements_to_read_this_iteration = std::min(elements_remaining, MAX_NUMBER_OF_ELEMENTS);
+      auto bytes_to_read = elements_to_read_this_iteration * uint32_t_size;
+      auto bytes_read_this_iteration =
+          static_cast<size_t>(read(fd, read_data_start + from + elements_read, bytes_to_read));
+      Assert((bytes_read_this_iteration == bytes_to_read),
+             close_file_and_return_error_message(fd, "Read error: ", errno));
+      elements_read += elements_to_read_this_iteration;
+      elements_remaining -= elements_to_read_this_iteration;
+    }
+  } else {
```

Comment:
Is this else really necessary?
I think the first if branch would also work when the workload is small enough for one read call.

Reply:
Same for the other methods.

Reply:
Yes, but is it still comparable then? Why should we use some overhead logic in a "fair" measurement if this overhead (iteration) is not necessary?